crates/bevy_color_blindness/src/lib.rs

```diff
@@ -0,0 +1,399 @@
+    #[uniform(2)]
+    green: Vec3,
+    #[uniform(2)]
+    blue: Vec3,
+}
+
+impl Material2d for PostProcessingMaterial {
+    fn fragment_shader() -> ShaderRef {
+        ShaderRef::Handle(COLOR_BLINDNESS_SHADER_HANDLE.typed())
+    }
+}
+
+/// Component to identify your main camera
+///
+/// Adding this component to a camera will set up the post-processing pipeline
+/// which simulates color blindness. This is done by changing the render target
+/// to be an image, and then using another camera to render that image.
+///
+/// Cameras with `ColorBlindnessCamera` will have [`UiCameraConfig`] inserted with
+/// `show_ui` set to `false`. This is to ensure that UI elements are not rendered twice.
+/// In most cases, you will want to render UI using the final post-processing camera.
+/// If for some reason this behavior is not desired, please open an issue.
+///
+/// [`UiCameraConfig`]: bevy_ui::entity::UiCameraConfig
+#[derive(Component)]
+pub struct ColorBlindnessCamera;
+
+/// sets the target for newly added `ColorBlindCamera`s
+fn set_camera_target(
+    mut commands: Commands,
+    mut query: Query<(Entity, &mut Camera), Added<ColorBlindnessCamera>>,
+    inner: Res<InternalResource>,
+) {
+    for (entity, mut camera) in query.iter_mut() {
+        camera.target = RenderTarget::Image(inner.image.clone());
+        commands
+            .entity(entity)
+            .insert(UiCameraConfig { show_ui: false });
+    }
+}
+
+/// updates the percentages in the post processing material when the Mode changes in Params
+fn update_percentages(
+    params: Res<ColorBlindnessParams>,
+    inner: Res<InternalResource>,
+    mut materials: ResMut<Assets<PostProcessingMaterial>>,
+) {
+    if params.is_changed() {
+        let mut mat = materials.get_mut(&inner.post).unwrap();
+
+        let mode = if params.enable {
+            &params.mode
+        } else {
+            &Mode::Normal
+        };
+        let (red, green, blue) = mode.percentages();
+
+        mat.red = red;
+        mat.green = green;
+        mat.blue = blue;
+    }
+}
+
+/// internal resource which holds the handles
+struct InternalResource {
+    image: Handle<Image>,
+    post: Handle<PostProcessingMaterial>,
+}
+
+/// creates the image, the material, the final camera, and the whole post-processing pipeline
+///
+/// based on the post-processing example
+/// https://github.com/bevyengine/bevy/blob/main/examples/shader/post_processing.rs
+fn setup(
+    mut commands: Commands,
+    mut windows: ResMut<Windows>,
+    mut meshes: ResMut<Assets<Mesh>>,
+    mut post_processing_materials: ResMut<Assets<PostProcessingMaterial>>,
+    mut images: ResMut<Assets<Image>>,
+    asset_server: Res<AssetServer>,
+    params: Res<ColorBlindnessParams>,
+) {
+    asset_server.watch_for_changes().unwrap();
+
+    let window = windows.get_primary_mut().unwrap();
+    let size = Extent3d {
+        width: window.physical_width(),
+        height: window.physical_height(),
+        ..Default::default()
```

Comment:
This feels highly suspect on a cursory reading.

If at all possible, we should avoid specialising to the primary window.

I don't really know how postprocessing is meant to work, but I doubt it's idiomatic for it to depend on windowing directly.

Reply:
I agree. I'm not too well versed with how multiple/no windows works on Bevy, I'll look into it a bit and come up with something that doesn't rely on a primary window existing.